my_compassion/static/src/js/write_a_letter.js

```diff
@@ -264,97 +247,56 @@ function startStopLoading(type) {
-  };
-  let request = new Request(
-    `${window.location.origin}/mobile-app-api/correspondence/get_preview`
-  );
-  let response = await fetch(request, init);
-  return response.text().then(function (text) {
-    if (with_loading) {
-      startStopLoading("preview");
-    }
-    if (!response.ok) {
-      displayAlert("preview_error");
-      return false;
-    }
-    if (preview) {
-      window.open(text.slice(1, -1), "_blank");
-    }
-    return true;
+  // Send the json data to odoo using a post request
+  $.ajax({
+    url: `${window.location.origin}/my/letter/${params.get("child_id")}/${mode}`,
+    type: "POST",
+    contentType: "application/json",
+    data: JSON.stringify(json_data),
+    success: function (response, status) {
+      startStopLoading(mode);
+      if (status !== "success" || response["error"] !== undefined) {
+        displayAlert(`${mode}_error`);
+        return;
+      }
+      if (mode === "preview") {
+        sessionStorage.setItem("generator_id", response.result["generator_id"]);
+        window.open(response.result["preview_url"], "_blank");
+      } else if (mode === "send") {
+        // Empty images and text (to avoid duplicate)
+        letter_content.value = "";
+        images_list.forEach((image) => {
+          removeImage(image.name, image.size, image.type);
+        });
+        $("#letter_sent_correctly").modal("show");
+        $(".christmas_action").toggleClass("d-none");
+        sessionStorage.removeItem("generator_id");
+      }
+    },
+    error: function (error) {
+      startStopLoading(mode);
+      displayAlert(`${mode}_error`);
+    },
   });
 }
 
-/**
- * This function takes care of sending a letter when the corresponding button
- * is clicked
- */
-async function sendLetter() {
-  startStopLoading("sending");
-  await createLetter((preview = false), (with_loading = false));
-
-  let params = new URLSearchParams(window.location.search);
-
-  let json_data = {
-    TemplateID: params.get("template_id"),
-    Need: params.get("child_id"),
-  };
-
-  params.forEach(function (v, k, _) {
-    json_data[k] = v;
-  });
-
-  let init = {
-    method: "POST",
-    headers: new Headers({ "Content-Type": "application/json" }),
-    body: JSON.stringify(json_data),
-  };
-  let request = new Request(
-    `${window.location.origin}/mobile-app-api/correspondence/send_letter`
-  );
-  let response = await fetch(request, init);
-  console.log(response);
-  let answer = response.text().then(function (text) {
-    if (!response.ok) {
-      displayAlert("letter_error");
-      return false;
-    }
-
-    // Empty images and text (to avoid duplicate)
-    letter_content.value = "";
-    for (let i = 0; i < images_list.length; i++) {
-      let image = images_list[i];
-      removeImage(image.name, image.size, image.type);
-    }
-
-    $("#letter_sent_correctly").modal("show");
-    $(".christmas_action").toggleClass("d-none");
-  });
-  startStopLoading("sending");
+function displayAlert(id) {
```

Comment:
This function displayAlert is duplicated in my_compassion/static/src/js/my_children.js:54 
But as we talked, no need to take too much time on this code as we will get rid of it in version 2.0